src/ml_gym/frontend/dashboard/src/features/jobsStatus/jobsStatusSlice.tsx

```diff
@@ -0,0 +1,50 @@
+import { createSlice } from '@reduxjs/toolkit'
+import { JobStatusMessageCollectionType, JobStatusType, JobStatusRowType} from '../../app/datatypes'
+import type { RootState } from '../../app/store';
+
+
+
+const initialState = {
+    messages: [],
+    job_id_to_latest_message_index: {}
+} as JobStatusMessageCollectionType
+
+
+const jobsStatusSlice = createSlice({
+    name: 'jobsStatus',
+    initialState,
+    reducers: {
+        jobStatusAdded(state, action) {
+            const job_id: number = action.payload.data.payload.job_id
+            if (job_id in state.job_id_to_latest_message_index) {
+                const row_id = state.job_id_to_latest_message_index[job_id]
+                // prevents possible race condition
+                if (state.messages[row_id].event_id < action.payload.event_id) {
+                    state.messages[row_id] = action.payload
+                }
+            } else {
+                state.messages.push(action.payload)
+                state.job_id_to_latest_message_index[job_id] = state.messages.length - 1
+            }
+        }
+    }
+})
+
+export const { jobStatusAdded } = jobsStatusSlice.actions
+
+export const jobStatusRowsSelector = (state: RootState) => state.jobsStatus.messages.map((s: JobStatusType) => (
+    {
```

Comment:
Codacy has a fix for the issue: [Insert `··(`](https://app.codacy.com/gh/mlgym/mlgym/pullRequest?prid=9966992)
```suggestion
      ({
```